### PR TITLE
131/components/uip 562 radio

### PR DESCRIPTION
**Pull request by akdetrick**
## Description

- Updates `Checkbox` to use uuid for `htmlFor` and `id` attributes
- Changes required props for `Checkbox`
- Adds `Radio`

## Screenshots
Screen Shot 2020-01-22 at 6 25 36 PM
Screen Shot 2020-01-22 at 6 18 55 PM
Screen Shot 2020-01-22 at 6 19 05 PM
Screen Shot 2020-01-22 at 6 19 19 PM
Screen Shot 2020-01-22 at 6 19 25 PM

## Checklist
- [x] Docs have been rebuilt (`yarn build:full`)
- [x] All unit tests pass
- [x] Snapshots have been updated
- [x] No lint errors or warnings have been introduced
- [x] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

File: 1 [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+  const handleChange = () => {
+    if (!disabled) {
+      const updatedCheckedState = !isChecked;
+      setIsChecked(updatedCheckedState);
+      onChange(updatedCheckedState);
+    }
+  };
+
+  let Icon = isChecked ? RadioFilledIcon : RadioEmptyIcon;
```
Comment (rey-wright) [inline 1]: const?

File: 1 [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  onChange: PropTypes.func,
+
+  /** Sets `checked` state */
+  checked: PropTypes.bool,
+
+  /** Disables form field when `true` */
+  disabled: PropTypes.bool,
+};
+
+export default Radio;
```
Comment (rey-wright) [inline 1]: Why is this filed called `1`? Sort of confused here.

**Comment by rey-wright** [comment]
Screen Shot 2020-01-22 at 6 34 31 PM

This is a silly nitpick but I feel like the notification explain that this spread props is just too big. I almost wonder if we can create a `StoryTag` component, without too much margin on it, and We can just add a "Spread Props" tag on it, and other components that spread props, if we feel it's necessary to add that here.

**Comment by rey-wright** [comment]
Do we really believe having these form fields use that uuid package is the right move? technically we could just let developers continue to supply their own id's... I'm sort of curious if that uuid package is invoking a big hit. This is just purely a curiosity question though.

**Comment by akdetrick** [comment]
uuid package is very tiny and likely to get tree shaken in cbi-site - I decided to go that route because it's necessary to link the label to the input for these to work correctly.

I believe that leaving it to developers to pass unique ids may lead to bugs

**Comment by rey-wright** [comment]
I'd like either of us (I volunteer if it makes sense) to take a FDS build of this, and try to replace some checkboxes on the site, and push it into staging... just to see where we stand at.

**Comment by rey-wright** [comment]
Outside of giving it a test run in `cbi-site`, and small documentation nitpicks, I really don't have too much further feedback on this one... looks good.

**Comment by akdetrick** [comment]
agreed on the size of the info box - it's a bit much, but not sure what to do to improve it yet

**Comment by rey-wright** [comment]
@akdetrick we can tackle it later for sure. I'd ignore it for now.

**Comment by akdetrick** [comment]
confirmed this works in `cbi-site`; going to resolve conflicts and merge

**Comment by rey-wright** [comment]
🚢

**Pull request merged**